src/main/java/org/generationcp/middleware/service/impl/inventory/LotServiceImpl.java

```diff
@@ -115,21 +119,66 @@ public String saveLot(final CropType cropType, final Integer userId, final LotGe
 	}
 
 	@Override
-	public void updateLots(final List<ExtendedLotDto> lotDtos, final LotUpdateRequestDto lotRequest) {
+	public void updateLots(final List<ExtendedLotDto> lotDtos, final LotUpdateRequestDto lotUpdateRequestDto) {
+		if (lotUpdateRequestDto.getSingleInput() != null) {
+			this.updateLots(lotDtos, lotUpdateRequestDto.getSingleInput());
+		} else {
+			this.updateLots(lotDtos, lotUpdateRequestDto.getMultiInput());
+		}
+	}
+
+	private void updateLots(final List<ExtendedLotDto> lotDtos, final LotSingleUpdateRequestDto lotSingleUpdateRequestDto){
 		final LotDAO lotDao = this.daoFactory.getLotDao();
 		for (final LotDto lotDto : lotDtos) {
 			final Lot lot = lotDao.getById(lotDto.getLotId());
```

Comment:
It would be better if we make only one call to the database to get all the lots you want to update and have a map by UUID with the lotDtos to find the new values